### PR TITLE
refactoring the view args away

### PR DESCRIPTION
**Pull request by ethn**
should be ready for merge now.

review in conjunction with https://github.com/decko-commons/decko/pull/193

### CODE REVIEW

File: mod/metrics/set/abstract/02_sort_and_filter.rb [hunk 1]
```diff
@@ -18,44 +18,36 @@ def paging_view
...
-  end
-
-  view :card_list_header do
-    ""
-  end
-
-  def item_card_from_row row
-    Card.fetch row[0]
-  end
+  # view :card_list_items do
```
Comment (ethn) [inline mod/metrics/set/abstract/02_sort_and_filter.rb]: I couldn't figure out how these views changed anything from default search behavior and couldn't see consequences for removing them (though it seemed undesirable to have them hanging around).  Did I miss something?

File: mod/metrics/set/right/downvotee_search.rb [hunk 2]
```diff
@@ -88,25 +88,9 @@ def list_with_session_votes
     }.freeze
   end
 
-  view :drag_and_drop, cache: :never do |args|
```
Comment (ethn) [inline mod/metrics/set/right/downvotee_search.rb]: another I couldn't find current use for...

File: mod/metrics/set/right/metric_weights.rb [hunk 3]
```diff
@@ -21,27 +21,28 @@ def metrics_with_weight
 end
 
 format :html do
```
Comment (ethn) [inline mod/metrics/set/right/metric_weights.rb]: is anything in this file used?

File: mod/wikirate/set/all/export.rb [hunk 4]
```diff
@@ -1,50 +0,0 @@
-format :json do
```
Comment (ethn) [inline mod/wikirate/set/all/export.rb]: most of this was already a duplicate of decko code.  new refactoring obviates it further.

File: mod/layout/set/abstract/listing.rb [hunk 5]
```diff
@@ -7,21 +7,16 @@
     wrap { haml :expanded_listing }
   end
 
-  view :listing_left do
-    "override listing_left view"
+  def self.view_for_override viewname
+    view viewname do
+      "override #{viewname} view"
+    end
```
Comment (xithan) [inline mod/layout/set/abstract/listing.rb]: move to decko?

File: mod/metrics/set/right/downvotee_search.rb [hunk 6]
```diff
@@ -201,36 +168,13 @@ def company_draggable_opts votee, opts
...
-          wrap_with(:div, class: "alert alert-info unsaved-message") do
-            args[:unsaved]
-          end
-        end,
-        yield
-      ].compact.join.html_safe
-    end
-  end
-
   def draggable content, args
```
Comment (xithan) [inline mod/metrics/set/right/downvotee_search.rb]: If you remove the drag&drop code you should remove this one, too

File: mod/metrics/set/right/metric_weights.rb (hunk 3 above)
Comment (xithan) [reply mod/metrics/set/right/metric_weights.rb]: the codename `metric_weights` doesn't exist. I don't remember if it's a feature that was never finished or is handled somewhere else

**Review by xithan — APPROVED** [review]

**Pull request merged**